nexus/types/src/deployment/planning_input.rs

```diff
@@ -173,7 +173,7 @@ impl PlanningInput {
     pub fn all_sleds(
         &self,
         filter: SledFilter,
-    ) -> impl Iterator<Item = (SledUuid, &SledDetails)> + '_ {
```

Comment:
huh, what made this Clone requirement show up?

Reply:
https://github.com/oxidecomputer/omicron/pull/6973/files#diff-3d648e59b20ff006cd1642fd0572f90c55a3106cebb8288eab9581b19ff38e78R338-R346

If github doesn't like that link: the `DisksEditor::build()` method calls `into_*_map` for both disks and datasets, both of which want to iterate over the sled IDs.